finality-aleph/src/justification/requester.rs

```diff
@@ -16,6 +16,64 @@ use crate::{
     network, Metrics,
 };
 
+/// This structure is created for keeping and reporting status of BlockRequester
+pub struct LastRequest<B: BlockT> {
+    block_number: Option<NumberFor<B>>,
+    block_hash: Option<B::Hash>,
+    tries: u64,
+}
+
+impl<B: BlockT> LastRequest<B> {
+    fn new() -> Self {
+        Self {
+            block_number: None,
+            block_hash: None,
+            tries: 0,
+        }
+    }
+
+    fn save_block_number(&mut self, num: NumberFor<B>) {
+        if Some(num) == self.block_number {
+            self.tries += 1;
+        } else {
+            self.block_number = Some(num);
+            self.block_hash = None;
+            self.tries = 1;
+        }
+    }
+
+    fn insert_hash(&mut self, hash: B::Hash) {
+        self.block_hash = Some(hash);
+    }
+
+    fn try_reset(&mut self, number: NumberFor<B>) {
```

Comment:
What's the purpose of this reset?

Reply:
So that we don't log for example 3 retries for block after we received it, but before we request another block number.

Reply:
But why would that be wrong? 
We are logging info about this particular component, and don't care what happened outside. Actually the logs are here mostly because there might be bugs when interacting with outside components. In particular, if we are logging info about requests, let's focus on requests. Whether we received something is a different matter.

Reply:
Ok, good point. I will remove it then.